tensorflow_quantum/datasets/spin_system.py

```diff
@@ -0,0 +1,240 @@
+# Copyright 2020 The TensorFlow Quantum Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Quantum datasets for several quantum many-body spin systems in the form of
+parameterized circuits"""
+from typing import List
```

Comment:
I do appreciate the use of typing, but unlike Cirq we don't type check our code and to keep things consistent it might be better if we removed it for now.